pkg/api/graph/deployment.go

```diff
@@ -21,8 +21,9 @@ type DeploymentPipelineMap map[*DeploymentConfigNode][]ImagePipeline
 // ImagePipeline represents a build, its output, and any inputs. The input
 // to a build may be another ImagePipeline.
 type ImagePipeline struct {
-	Image ImageTagLocation
-	Build *BuildConfigNode
+	Image       ImageTagLocation
+	ImageStream *ImageStreamNode
```

Comment:
So would I, but that means that you end up with a method on a shared interface that's only sometimes valid.  That's your preference?

Also, why do we have the veneer on the graph.  Straight graph traversal would make this point moot.